packages/ember-metal/lib/property_get.js

```diff
@@ -52,11 +52,22 @@ export function get(obj, keyName) {
   assert(`The key provided to get must be a string, you passed ${keyName}`, typeof keyName === 'string');
   assert(`'this' in paths is not supported`, !pathHasThis(keyName));
 
-  // Helpers that operate with 'this' within an #each
-  if (keyName === '') {
+  let type = typeof obj;
+  let isObject = obj !== null && type === 'object';
+  let isFunc = type === 'function';
+  let isStringLength = type === 'string' && keyName === 'length';
+
+  if (isObject || isFunc || isStringLength) {
+    return _get(obj, keyName);
+  } else if (keyName === '') {
```

Comment:
yeah, the if helper's falsy is the primary culprit for get(bool, 'length') I'm cool with just focusing on that and getting it in and cleaning up the other stuff separately, it will be easier to backport if we want to do that.  @stefanpenner seem good?